File: ros/test_cpp/src/chat/src/listener.cpp
#include <memory>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

using std::placeholders::_1;

class Listener : public rclcpp::Node
{
public:
  Listener()
  : Node("listener")
  {
    subscription =
      this->create_subscription<std_msgs::msg::String>(
      "chat", 10,
      std::bind(&Listener::topic_callback, this, _1));
  }

private:
  void topic_callback(const std_msgs::msg::String::SharedPtr msg) const
  {
    RCLCPP_INFO(this->get_logger(), "I heard: '%s'", msg->data.c_str());
  }

  rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<Listener>());
  rclcpp::shutdown();

  return 0;
}

File: experiments/test_js/src/listener.ts
import ros, { std_msgs } from "rclnodejs";

ros.init().then(() => {
  const node = new ros.Node("listener");

  node.createSubscription(
    "std_msgs/msg/String",
    "chat",
    {},
    (message: std_msgs.msg.String) => {
      console.log(`Received: ${message.data}`);
    }
  );

  node.spin();
});
